cumulus_library_umls/ancillary_tables.sql

```diff
@@ -0,0 +1,428 @@
+    tty,
+    code,
+    str,
+    3 AS depth
+FROM umls__icd10_block
+UNION ALL
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code,
+    str,
+    4 AS depth
+FROM umls__icd10_category
+UNION ALL
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code,
+    str,
+    4 AS depth
+FROM umls__icd10_subcategory_1
+UNION ALL
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code,
+    str,
+    5 AS depth
+FROM umls__icd10_subcategory_2
+UNION ALL
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code,
+    str,
+    6 AS depth
+FROM umls__icd10_subcategory_3
+UNION ALL
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code,
+    str,
+    7 AS depth
+FROM umls__icd10_extension;
+
+CREATE TABLE umls__icd10_hierarchy AS
+WITH chapter AS (
+    SELECT
+        code AS chapter_code,
+        str AS chapter_str,
+        cui2 AS cui
+    FROM umls__icd10_tree
+    WHERE depth = 2
+),
+
+block AS (
+    SELECT
+        p.chapter_code,
+        p.chapter_str,
+        c.code AS block_code,
+        c.str AS block_str,
+        c.cui2 AS cui
+    FROM chapter AS p
+    LEFT JOIN umls__icd10_tree AS c
+        ON p.cui = c.cui1
+),
+
+category AS (
+    SELECT
+        p.chapter_code,
+        p.chapter_str,
+        p.block_code,
+        p.block_str,
+        c.code AS category_code,
+        c.str AS category_str,
+        c.cui2 AS cui
+    FROM block AS p
+    LEFT JOIN umls__icd10_tree AS c
+        ON p.cui = c.cui1
+),
+
+subcategory_1 AS (
+    SELECT
+        p.chapter_code,
+        p.chapter_str,
+        p.block_code,
+        p.block_str,
+        p.category_code,
+        p.category_str,
+        c.code AS subcategory_1_code,
+        c.str AS subcategory_1_str,
+        c.cui2 AS cui
+    FROM category AS p
+    LEFT JOIN umls__icd10_tree AS c
+        ON p.cui = c.cui1
+),
+
+subcategory_2 AS (
+    SELECT
+        p.chapter_code,
+        p.chapter_str,
+        p.block_code,
+        p.block_str,
+        p.category_code,
+        p.category_str,
+        p.subcategory_1_code,
+        p.subcategory_1_str,
+        c.code AS subcategory_2_code,
+        c.str AS subcategory_2_str,
+        c.cui2 AS cui
+    FROM subcategory_1 AS p
+    LEFT JOIN umls__icd10_tree AS c
+        ON p.cui = c.cui1
+        -- From here on out, we need to filter out some circular refs in UMLS,
+        -- so we'll start looking for codes of specified lengths
+        AND length(c.code) = 6
+),
+
+subcategory_3 AS (
+    SELECT
+        p.chapter_code,
+        p.chapter_str,
+        p.block_code,
+        p.block_str,
+        p.category_code,
+        p.category_str,
+        p.subcategory_1_code,
+        p.subcategory_1_str,
+        p.subcategory_2_code,
+        p.subcategory_2_str,
+        c.code AS subcategory_3_code,
+        c.str AS subcategory_3_str,
+        c.cui2 AS cui
+    FROM subcategory_2 AS p
+    LEFT JOIN umls__icd10_tree AS c
+        ON
+            p.cui = c.cui1
+            AND length(c.code) = 7
+)
```

Comment:
Either this sequence of tables doesn't make sense or my brain is broken this morning - let's chat about what these joins are doing.